client/src/main/scala/org/http4s/client/middleware/GZip.scala

```diff
@@ -40,7 +40,7 @@ object GZip {
           req.headers ++ Headers.of(Header(`Accept-Encoding`.name.value, supportedCompressions)))
     }
 
-  @silent("deprecated")
+  @nowarn("msg=deprecated")
```

Comment:
I personally found that `@nowarn("cat=deprecation")` looks clearer since it disables warnings by the entire category rather than trying to figure it out from warning messages. Either way should work though, just a suggestion.

Reply:
...but I haven't tried it on Scala other than 2.13 though...

Reply:
I'm learning about this on the fly, but I was uncomfortable with what I had to do.  Let's try that.

Reply:
Silencer on 2.12 only supports `msg`, nothing else

Reply:
Yeah, indeed. According to their docs:
> NOTE: @nowarn in Scala 2.13.2 supports various fine-grained filters (e.g. warning category, message patttern, etc.). Silencer only supports the msg=<pattern> filter - all other filters simply suppress everything, as if there were no filters specified.

But perhaps it is not that bad in our case – looks like the `cat` filter works properly in Scala 2.13 and any meaningful warning is going to be caught up anyway.

Reply:
The 2.12 builds compiled without a fatal warning.  I think the silencer integration still did work?

Reply:
Looks like if the silencer cannot recognize a filter in `@nowarn`, it will simply suppress any kind of warnings for the annotated item.

Reply:
Oh, I see.  I think we can live with that behavior: we're talking about useful warnings that are found by 2.12, but not by 2.13 or soon Dotty, from places that already have a deprecation warning that we're ignoring.  That seems exceedingly unlikely.  The warnings are generally going to be more insightful on the newer compilers anyway.